### PR TITLE
Upload PaaS db_dump to Az storage_account

### PR DESCRIPTION
**Pull request by felixtheflex**
### Context

api-migration-store-prod-db-backups-in-azure

### Changes proposed in this pull request

Take a copy a dump from prod (before) sanitisation and store on Azure storage Account

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by felixtheflex** [comment]
[API migration - Store prod DB backups in Azure](https://trello.com/c/Pp8l1qOV/1169-api-migration-store-prod-db-backups-in-azure)

File: .github/workflows/database-restore.yml [hunk 1]
```diff
@@ -34,12 +34,34 @@ jobs:
          INSTALL_CONDUIT: true
 
       - name: Backup Teacher Training API Prod Database
-        run: cf conduit teacher-training-api-postgres-prod -- pg_dump --encoding utf8 --clean --no-owner --if-exists -f prod_backup.sql
-      
+        run: |
+          now=$(date +"%S_%M_%H_%m_%d_%Y")
+          prod_sql_file=prod_backup-$now.sql
+          touch $prod_sql_file
```
Comment (vigneshmsft) [inline .github/workflows/database-restore.yml:39]: `touch` isn't required

File: .github/workflows/database-restore.yml [hunk 2]
```diff
@@ -34,12 +33,34 @@ jobs:
...
+        with:
+          keyvault: "s121p01-shared-kv-01"
+          secrets: 'TTAPI-STORAGE-ACCOUNT-CONNECTION-STRING-PRODUCTION'
+        id: myGetSecretAction
+
+      - uses: bacongobbler/azure-blob-storage-upload@v1.1.1
+        with:
+          source_dir: "./"
+          container_name: 'sqldbdump'
+          connection_string: ${{ steps.myGetSecretAction.outputs.TTAPI-STORAGE-ACCOUNT-CONNECTION-STRING-PRODUCTION }}
```
Comment (vigneshmsft) [inline .github/workflows/database-restore.yml:56]: change to `getKeyVaultSecret` or similar?
remove `my` :)

File: .github/workflows/database-restore.yml [hunk 3]
```diff
@@ -34,12 +33,34 @@ jobs:
...
       - name: Backup Teacher Training API Prod Database
-        run: cf conduit teacher-training-api-postgres-prod -- pg_dump --encoding utf8 --clean --no-owner --if-exists -f prod_backup.sql
-      
+        run: |
+          now=$(date +"%S_%M_%H_%m_%d_%Y")
+          prod_sql_file=prod_backup-$now.sql
+          touch $prod_sql_file
+          cf conduit teacher-training-api-postgres-prod -- pg_dump --encoding utf8 --clean --no-owner --if-exists -f $prod_sql_file
+
+      - uses: azure/login@v1
```
Comment (vigneshmsft) [inline .github/workflows/database-restore.yml:42]: pls add `name:` to this action and also to the get-keyvault-secrets action

**Review by vigneshmsft — COMMENTED** [review]
nice, two minor comments..

File: .github/workflows/database-restore.yml (hunk 2 above)
Comment (felixtheflex) [reply .github/workflows/database-restore.yml:56]: lol

File: .github/workflows/database-restore.yml [hunk 4]
```diff
@@ -34,12 +35,37 @@ jobs:
...
+          cf conduit teacher-training-api-postgres-prod -- pg_dump --encoding utf8 --clean --no-owner --if-exists -f $prod_sql_file
+
+
+      - name: Login to Azure
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS_PRODUCTION }}
+
+      - name: Get storageAccount connection string from keyvault
+        uses: Azure/get-keyvault-secrets@v1.0
```
Comment (vigneshmsft) [inline .github/workflows/database-restore.yml:50]:
```suggestion
        uses: Azure/get-keyvault-secrets@v1
```

File: .github/workflows/database-restore.yml [hunk 5]
```diff
@@ -34,12 +35,37 @@ jobs:
          INSTALL_CONDUIT: true
 
       - name: Backup Teacher Training API Prod Database
-        run: cf conduit teacher-training-api-postgres-prod -- pg_dump --encoding utf8 --clean --no-owner --if-exists -f prod_backup.sql
-      
+        run: |
+          now=$(date +"%S_%M_%H_%m_%d_%Y")
+          prod_sql_file=prod_backup-$now.sql
```
Comment (vigneshmsft) [inline .github/workflows/database-restore.yml:40]:
```suggestion
          prod_sql_file=prod_backup-$now.sql
          echo "PROD_BACKUP=$prod_sql_file" >> $GITHUB_ENV
```
and use `${{ env.PROD_BACKUP }}` in the below steps

**Review by vigneshmsft — APPROVED** [review]

**Pull request merged**